articles/getting-started-with-jwt-using-angular8-and-nodejs/index.md

```diff
@@ -0,0 +1,271 @@
+
+## Creating the Backend
+
+I will start by creating a directory in the base folder for the server and initialize it as a node project by running the following commands.
+```bash
+$ mkdir node_server
+$ cd node_server
+$ npm init –y
+```
+The necessary libraries ([express](https://www.npmjs.com/package/express), [body-parser](https://www.npmjs.com/package/body-parser), [express-jwt](https://www.npmjs.com/package/express-jwt)) are required to be installed. Install them by running the command below:
+```bash
+$ npm i -S express cors body-parser express-jwt jsonwebtoken
+```
+After running the command above, a new file `app.js` containing the code below is created.
+```javascript
+const express       = require('express')
+const bodyParser    = require('body-parser');
+const cors          = require('cors');
+
+const app           = express();
+const port          = 3000;
+
+app.use(cors());
+app.options('*', cors());
+app.use(bodyParser.json({limit: '10mb', extended: true}));
+app.use(bodyParser.urlencoded({limit: '10mb', extended: true}));
+
+app.get('/', (req, res) => {
+    res.json("Hello World");
+});
+
+/* more code to be added later */
+
+
+/* the listen function */
+app.listen(port, function() {
+    console.log("Listening to " + port);
+});
+```
+Next, I will create a route where a token is generated, and it will be where the user authenticates. On successful authentication, the token is sent. Below code is added to the file `app.js`.
+```javascript
+//SECRET FOR JWT
+let secret = 'some_secret';
+
+/* Create token to be used */
+app.get('/token/sign', (req, res) => {
+    var userData = {
+        "name": "My Name",
+        "id": "1234"
+    }
+    let token = jwt.sign(userData, secret, { expiresIn: '15s'})
+    res.status(200).json({"token": token});
+});
+```
+Once the route has been created, it stores the user data and encodes the data. When decoded, it will give back the user data, in this case only username and id, since storing a password is never a good practice. Next will be to run the application and check the token generated by running:
+```bash
+$ node app.js
+```
+In this guide, I will use Postman to test the routes as shown below.
+
+![first web token](/engineering-education/getting-started-with-jwt-using-angular8-and-nodejs/token-generation1.png)
+
+As of now, I have generated the first web token. To illustrate this, I will create a `/path1` and secure it using my JSON Web Token.
+
+For this task, I will be using the express-jwt function.
+```javascript
+app.get('/path1', (req, res) => {
+    res.status(200)
+        .json({
+            "success": true,
+            "msg": "Secrect Access Granted"
+        });
+});
+```
+Using expressJWT function:
+```javascript
+//This is to allow access to the path with no token authentication
+app.use(expressJWT({ secret: secret})
+    .unless(
```

Comment:
This line throws a `ReferenceError: expressJWT is not defined` error. Looks like you forgot to import `express-jwt`.